csrc/expr_simplifier.cpp

```diff
@@ -1553,6 +1573,26 @@ bool lessThan(Val* x, Val* y, const Context& context) {
       return true;
     }
   }
+  for (const auto& [a, b] : context.getKnownLessEqual()) {
+    if (depth >= max_recursion_depth) {
+      // Limit the recursion depth to avoid infinite recursion
+      // In practice only a few levels are usually enough transitivity to prove
+      // what is needed for index expressions.
+      continue;
```

Comment:
Should this be `break`?

Reply:
Yeah you're right. Actually we should just move this check before the for loop.